geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/CacheClientProxy.java

```diff
@@ -774,14 +770,23 @@ protected boolean close(boolean checkQueue, boolean stoppedNormally) {
 
     connected = false;
 
-    // Close the Authorization callback (if any)
+    // Close the Authorization callback or subject if we are not keeping the proxy
     try {
       if (!pauseDurable) {
-        if (postAuthzCallback != null) {// for single user
+        // single user case -- old security
+        if (postAuthzCallback != null) {
           postAuthzCallback.close();
           postAuthzCallback = null;
         }
-        if (clientUserAuths != null) {// for multiple users
+        // single user case -- integrated security
+        // connection is closed, so we can log out this subject
+        else if (subject != null) {
+          logger.debug("CacheClientProxy.close, logging out: " + subject.getPrincipal());
```

Comment:
This is probably exposing sensitive data to the log even though it's not the password. The User can set the log-level to debug which turns on these log statements.

Also, all security related logging in the product should use the SECURITY_LOGGER_NAME:
```
private static final Logger logger = LogService.getLogger(SECURITY_LOGGER_NAME);
```
The SECURITY_LOGGER_NAME gets redirected to the `ConfigurationProperties.SECURITY_LOG_FILE` if the user has set it.

Don't forget, the log4j2 uses `{}` in the syntax:
```
logger.debug("CacheClientProxy.close, logging out: {}", subject.getPrincipal());
```

Reply:
done